peer/peerstorage.go

```diff
@@ -0,0 +1,103 @@
+package peer
+
+import (
+	"errors"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/lightningnetwork/lnd/kvdb"
+)
+
+var (
+	// peerStorage is the kvdb key where we store other peer's backup data
+	// which they send us.
+	peerStorage = []byte("peer-storage")
+
+	// ErrUninitializedDB is called when the DB is not initialized yet
+	// but a caller calls the `Retrieve` function.
+	ErrUninitializedDB = errors.New("uninitialized Kvdb peer data " +
+		"store")
+)
+
+// KvdbPeerStorage is the kvdb implementation of the PeerDataStore interface.
+type KvdbPeerStorage struct {
+	// DB is th kvdb backend for the storage layer.
+	DB kvdb.Backend
+}
+
+// NewKvdbPeerStorage returns an instance of the kvdb storage layer.
+func NewKvdbPeerStorage(dB kvdb.Backend) (*KvdbPeerStorage, error) {
+	err := kvdb.Update(dB, func(tx kvdb.RwTx) error {
+		_, err := tx.CreateTopLevelBucket(peerStorage)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}, func() {})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &KvdbPeerStorage{
+		DB: dB,
+	}, nil
+}
+
+// Store persists the peer's backup in the storage layer.
+func (k *KvdbPeerStorage) Store(data []byte,
+	peerPub btcec.PublicKey) error {
+
+	return kvdb.Update(k.DB, func(tx kvdb.RwTx) error {
+		bucket, err := tx.CreateTopLevelBucket(peerStorage)
+		if err != nil {
+			return err
+		}
```

Comment:
no need to call `CreateTopLevelBucket` again, that has already been called in `NewKvdbPeerStorage`. 

This should instead fetch the bucket as is done in `Retrieve` and return the uninitialised error if it is not found